### PR TITLE
Add High Level Memory Threshold Metrics

### PR DESCRIPTION
**Pull request by SpriCoder**
Add High-Level Memory Threshold Metrics(Total, Storage Engine, Query Engine, Schema Engine, Consensus, Stream Engine, Direct Buffer)

### CODE REVIEW

**Comment by SpriCoder** [comment]
Panel:
image

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/metrics/memory/ThresholdMemoryMetrics.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        "1");
+    metricService.remove(
+        MetricType.GAUGE,
+        Metric.THRESHOLD_MEMORY_SIZE.toString(),
+        Tag.NAME.toString(),
+        "DirectBuffer",
+        Tag.TYPE.toString(),
+        "OffHeap",
+        Tag.LEVEL.toString(),
+        "1");
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/metrics/memory/ThresholdMemoryMetrics.java:181]: You could put define some field for these magic level string

Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/metrics/memory/ThresholdMemoryMetrics.java:181]: You could define some field for these magic level string，could do it in next pr

**Review by OneSizeFitsQuorum — COMMENTED** [review]
LGTM

**Pull request merged**